main/include/chat/chat_client.hpp

```diff
@@ -0,0 +1,180 @@
+#include <assert.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <algorithm>
+#include <cctype>
+#include <chrono>
+#include <map>
+#include <mutex>
+#include <queue>
+#include <random>
+#include <string>
+#include <thread>
+
+#include "chat/connection_manager.h"
+#include "steam/steamnetworkingsockets.h"
+
+#pragma once
+
+using namespace std;
+using namespace std::chrono;
+
+template <class T>
+class ActiveObject {
```

Comment:
A suuuper basic passthrough active-object that can be used to manage a fire and forget queue in rxcpp